src/bp-document/bp-document-filters.php

```diff
@@ -756,3 +756,489 @@ function bp_document_activity_update_document_privacy( $activity ) {
 		}
 	}
 }
+
+/**
+ * Protect downloads from ms-files.php in multisite.
+ *
+ * @param string $rewrite rewrite rules.
+ * @return string
+ * @since BuddyBoss 1.4.1
+ */
+function bp_document_protect_download_rewite_rules( $rewrite ) {
+	if ( ! is_multisite() ) {
+		return $rewrite;
+	}
+
+	$rule  = "\n# Document Rules - Protect Files from ms-files.php\n\n";
+	$rule .= "<IfModule mod_rewrite.c>\n";
+	$rule .= "RewriteEngine On\n";
+	$rule .= "RewriteCond %{QUERY_STRING} file=document_uploads/ [NC]\n";
+	$rule .= "RewriteRule /ms-files.php$ - [F]\n";
+	$rule .= "</IfModule>\n\n";
+
+	return $rule . $rewrite;
+}
+add_filter( 'mod_rewrite_rules', 'bp_document_protect_download_rewite_rules' );
+
+function bp_document_check_download_folder_protection() {
+
+	$upload_dir     = wp_get_upload_dir();
+	$downloads_path = $upload_dir['basedir'] . '/bb_documents';
+	$file_path      	= $downloads_path . '/.htaccess';
+	if ( false === ( $value = get_transient( 'bp_document_transient_htaccess' ) ) || ! file_exists( $file_path ) ) {
+
+		$file_content    = 'deny from all
```

Comment:
This line does not allow downloading files.
Details: https://github.com/buddyboss/buddyboss-platform/issues/987#issuecomment-640288019